lib/shopify_api/resources/variant.rb

```diff
@@ -4,5 +4,36 @@ class Variant < Base
     include DisablePrefixCheck
 
     conditional_prefix :product
+
+    def inventory_quantity_adjustment=(new_value)
+      raise_deprecated_inventory_call('inventory_quantity_adjustment') unless allow_inventory_params?
+      super
+    end
+
+    def inventory_quantity=(new_value)
+      raise_deprecated_inventory_call('inventory_quantity') unless allow_inventory_params?
+      super
+    end
+
+    def old_inventory_quantity=(new_value)
+      raise_deprecated_inventory_call('old_inventory_quantity') unless allow_inventory_params?
+      super
+    end
+
+    def serializable_hash(options = {})
+      allow_inventory_params? ? super(options) : super(options).except('inventory_quantity', 'old_inventory_quantity')
```

Comment:
Is this accurate? Those fields were deprecated for writes, but reading `variant.total_inventory` should still be supported